library/md2.c

```diff
@@ -146,14 +148,19 @@ void mbedtls_md2_process( mbedtls_md2_context *ctx )
            ( ctx->cksum[i] ^ PI_SUBST[ctx->buffer[i] ^ t] );
         t  = ctx->cksum[i];
     }
+
+    return( 0 );
 }
 #endif /* !MBEDTLS_MD2_PROCESS_ALT */
 
 /*
  * MD2 process buffer
  */
-void mbedtls_md2_update( mbedtls_md2_context *ctx, const unsigned char *input, size_t ilen )
+int mbedtls_md2_update_ext( mbedtls_md2_context *ctx,
+                            const unsigned char *input,
+                            size_t ilen )
 {
+    int ret;
```

Comment:
in case ilen is zero, ret will not be assigned. There is a possibility that some static analyzers will throw a warning on this variable not being initialized

Reply:
@RonEld: Thanks for the review. I do not think the uninitialised variable is a problem because it is never used. Also, similar patterns occur in various places in the library. I think I will leave it like this unless there are strong objections